drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/conversion/GuidedDecisionTablePopulater.java

```diff
@@ -63,417 +66,485 @@
+                    setZeroParameterConditionColumnHeader(column,
+                                                          variableColumns);
                 } else {
-                    setCompositeColumnHeader( column );
+                    setCompositeColumnHeader(column);
                 }
             }
         }
 
-        if ( rm.rhs != null ) {
-            for ( IAction action : rm.rhs ) {
+        if (rm.rhs != null) {
+            for (IAction action : rm.rhs) {
                 final BRLActionColumn column = new BRLActionColumn();
-                column.getDefinition().add( action );
-                dtable.getActionCols().add( column );
+                column.getDefinition().add(action);
+                dtable.getActionCols().add(column);
+
+                if (action instanceof ActionRetractFact) {
+                    final ActionRetractFact arf = (ActionRetractFact) action;
+                    createActionRetractFactChildColumns(arf,
+                                                        column,
+                                                        variableColumns);
+                } else {
+                    createRHSChildColumns(rm,
+                                          action,
+                                          column,
+                                          variableColumns);
+                }
 
-                final Map<InterpolationVariable, Integer> templateKeys = new HashMap<>();
-                final RuleModelVisitor rmv = new RuleModelVisitor( rm.lhs,
-                                                                   templateKeys );
-                rmv.visit( action );
+                if (column.getChildColumns().isEmpty()) {
+                    setZeroParameterActionColumnHeader(column,
+                                                       variableColumns);
+                } else {
+                    setCompositeColumnHeader(column);
+                }
+            }
+        }
+    }
 
-                final List<InterpolationVariable> ivs = new ArrayList<>( templateKeys.keySet() );
-                for ( BRLVariableColumn variableColumn : variableColumns ) {
-                    final Iterator<InterpolationVariable> ivsIts = ivs.iterator();
-                    while ( ivsIts.hasNext() ) {
-                        final InterpolationVariable iv = ivsIts.next();
-                        if ( iv.getVarName().equals( variableColumn.getVarName() ) ) {
-                            final BRLActionVariableColumn source = (BRLActionVariableColumn) variableColumn;
-                            final BRLActionVariableColumn target = makeBRLActionVariableColumn( source,
-                                                                                                iv );
-                            column.getChildColumns().add( target );
-                            ivsIts.remove();
+    private void createActionRetractFactChildColumns(final ActionRetractFact arf,
```

Comment:
This is where the fun is. At this stage the XLS column has been converted to a ```ActionRetractFact``` ```IAction``` on the ```BRLActionColumn``` definition. However the "variable" is that of the XLS parameter (e.g. $param). This procedure checks whether the ```ActionRetractFact``` "variable" relates to a ```ParameterizedValueBuilder``` and if so updates the ```ActionRetractFact``` "variable" to the first value in the XLS data.